src/se/mod.rs

```diff
@@ -528,8 +528,19 @@ impl<'w, 'r, W: Write> Serializer<'w, 'r, W> {
     }
 
     /// Configure indent for a serializer
-    pub fn indent(&mut self, indent_char: char, indent_size: usize) -> &mut Self {
-        self.ser.indent = Indent::Owned(Indentation::new(indent_char as u8, indent_size));
+    ///
+    /// Note: `initial_indent_amt` is the initial total length of the indent, not the indent level
+    pub fn indent(
+        &mut self,
+        indent_char: char,
+        indent_size: usize,
+        initial_indent_amt: usize,
```

Comment:
I find this part a bit unsatisfying, I'd rather it be an initial "indent level" that is multiplied by the `indent_size` to get the initial indent.

Reply:
Agreed that this would be better.

I think is would be even better to create a new internal method to just set `Indent` on serializer directly. Then no need to introduce new parameter here.